### PR TITLE
DATAUP-564 app_name for batch job info

### PR DESCRIPTION
**Pull request by n1mus**
# Description of PR purpose/changes

The batch container job was previously being skipped when sending `job_info` requests BE->FE. That's because it couldn't quite fill the `"app_name"` field since it couldn't get an app spec due to its `app_id` being `"batch"`.

@ialarmedalien thought of a design where a batch container job's de facto `"app_name"` should be `"batch"`, thus allowing the batch container job to be included when sending `job_info` requests BE->FE. I.e., the `job` object's `app_name` attribute should be set to `"batch"` for batch container jobs.

`app_name` is not looked up in `__getattr__` because conventionally that's more for the attributes from the EE2 state. Also, `was_terminal` seems like it should be more an action than a `property` 

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [n/a] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [n/a] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: src/biokbase/narrative/jobs/jobcomm.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def _lookup_job_info_batch(self, req: JobRequest) -> dict:
         self._verify_job_id(req)
         try:
             job_ids = self._jm.update_batch_job(req.job_id)
-            job_info_batch = self._jm.lookup_job_info(job_ids[1:])
```
Comment (n1mus) [inline src/biokbase/narrative/jobs/jobcomm.py:298]: Was skipping batch container job before for `job_info_batch`

File: src/biokbase/narrative/tests/test_jobcomm.py [hunk 2]
```diff
@@ -44,35 +44,12 @@
     get_test_job_states,
     get_cell_2_jobs,
 )
+from .test_jobmanager import get_test_job_info, get_test_job_infos
 
 
 APP_NAME = "The Best App in the World"
 
 
-def get_test_job_infos(job_ids):
```
Comment (n1mus) [inline src/biokbase/narrative/tests/test_jobcomm.py:52]: Move to most upstream file `test_jobcomm.py`

File: src/biokbase/narrative/tests/test_jobcomm.py [hunk 3]
```diff
@@ -392,7 +369,7 @@ def test_lookup_job_states_batch__not_batch(self):
     # Lookup job info
     # -----------------------
     def test_lookup_job_info__ok(self):
-        job_id_list = [JOB_COMPLETED]
+        job_id_list = ALL_JOBS
```
Comment (n1mus) [inline src/biokbase/narrative/tests/test_jobcomm.py:372]: Include batch container job now (and a lot others)

File: src/biokbase/narrative/tests/test_jobcomm.py [hunk 4]
```diff
@@ -404,19 +381,6 @@ def test_lookup_job_info__ok(self):
             msg["data"],
         )
 
-    def test_lookup_job_info__batch__ok(self):
```
Comment (n1mus) [inline src/biokbase/narrative/tests/test_jobcomm.py:407]: This was testing that batch container job was skipped

File: src/biokbase/narrative/jobs/jobmanager.py [hunk 5]
```diff
@@ -288,20 +287,20 @@ def lookup_job_info(self, job_ids: List[str]) -> dict:
...
         infos = dict()
         for job_id in job_ids:
             job = self.get_job(job_id)
-            if not job.batch_job:
-                infos[job_id] = {
-                    "app_id": job.app_id,
-                    "app_name": job.app_spec()["info"]["name"],
-                    "job_id": job_id,
-                    "job_params": job.params,
-                    "batch_id": job.batch_id,
```
Comment (n1mus) [inline src/biokbase/narrative/jobs/jobmanager.py:303]: `"app_name"` and `"batch_id"` are artificially set for batch container jobs now

**Review by ialarmedalien — APPROVED** [review]
Looks great!

**Pull request merged**